File: java18_tree_bst/tree_bst/src/p4.java
// size of binary tree size()
// sum of nods of binary tree sum()
// muximum val in bt
// level of tree
public class p4 {
    static class d{
        static int size(Node a){
            if(a == null){
                return 0;
            }
            return 1+size(a.left)+size(a.right);
        }
        static int sum(Node a){
            if(a == null){
                return 0;
            }
            return a.val + sum(a.left)+sum(a.right);
        }
        static int maximum(Node a,int max){
            if(a == null){
                return 0;
            }
           int k =  maximum(a.left,max);
            int p =  maximum(a.right,max);
            int val =  Math.max(k,p);
             return Math.max(val, a.val);
        }
        static int level(Node a){
            if(a == null){
                return 0;
            }
            return 1 + Math.max(level(a.left),level(a.right));
        }
    }
    public static void main(String[] args) {
        Node a = new Node(1);

        Node b = new Node(2);
        Node c = new Node(31);

        Node d = new Node(4);
        Node e = new Node(5);
        Node f = new Node(6);
        Node g = new Node(7);

        a.right = c; a.left = b;
        b.right = e; b.left = d;
        c.right = g; c.left = f;

        d kp = new d();
        System.out.println(kp.size(a)); //7
        System.out.println(kp.sum(a)); //56
        System.out.println(kp.maximum(a,0)); //31
        System.out.println(kp.level(a));//3

    }
}

File: java20_tree_bst_p3/bst3/src/p4.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} val
 * @return {TreeNode}
 */


//  Insert into a Binary Search Tree

function ans(root,val){
  if(root.val>val) {
     if(root.left == null){
        let temp = new TreeNode(val);
        root.left = temp
        return;
     }
   ans(root.left,val);
  }
  if(root.val<val){
    if(root.right == null){
        let temp = new TreeNode(val);
        root.right = temp
        return;
     }
  ans(root.right,val);
  }
}
var insertIntoBST = function(root, val) {
    let temp = new TreeNode(val);
    if(root == null) return temp;
    ans(root,val);
    return root;
};